kernel/src/scan/state.rs

```diff
@@ -195,14 +199,15 @@ impl<T> RowVisitor for ScanFileVisitor<'_, T> {
                 let dv_info = DvInfo { deletion_vector };
                 let partition_values =
                     getters[9].get(row_index, "scanFile.fileConstantValues.partitionValues")?;
-                (self.callback)(
-                    &mut self.context,
-                    path,
+
+                let scan_file = ScanFile {
+                    path: path.to_string(),
```

Comment:
It's already a string
```suggestion
                    path,
```